File: src/etch/bindings/csharp/support/Pool.cs
using System;
using System.Text;
using System.Collections.Generic;

namespace Etch.Support
{
    /// <summary>
    /// Interface to a queued or free thread pool.
    /// </summary>
    public interface Pool
    {
        void Run( RunDelegate d1, ExceptionDelegate d2 );
    }

    /// <summary>
    /// Delegate for what to run
    /// </summary>
    public delegate void RunDelegate();

    /// <summary>
    /// Delegate that can throw an exception
    /// </summary>
    /// <param name="e"></param>
    public delegate void ExceptionDelegate( Exception e );
}

File: src/etch/bindings/java/support/Pool.java
/**
 * 
 */
package etch.bindings.java.support;

/**
 * Interface to a queued or free thread pool.
 */
public interface Pool
{
	/**
	 * @param runnable the thing to run.
	 * @throws Exception if there is a problem scheduling the
	 * runnable to run.
	 */
	void run( PoolRunnable runnable ) throws Exception;
	
	/**
	 * A runnable which can throw an exception.
	 */
	public interface PoolRunnable
	{
		/**
		 * @throws Exception
		 */
		public void run() throws Exception;

		/**
		 * Reports an exception thrown by the run method.
		 * @param e
		 */
		public void exception( Exception e );
	}
}
